### PR TITLE
Add SnapshotCodecV4 in cdap-default

### PR DESCRIPTION
**Pull request by rohitsinha54**
Build:http://builds.cask.co/browse/CDAP-DUT2762-2

### CODE REVIEW

File: cdap-common/src/main/resources/cdap-default.xml [hunk 1]
```diff
@@ -405,10 +405,11 @@
   <property>
     <name>data.tx.snapshot.codecs</name>
     <value>
-            co.cask.cdap.data2.transaction.snapshot.SnapshotCodecV1,
-            co.cask.cdap.data2.transaction.snapshot.SnapshotCodecV2,
-            co.cask.tephra.snapshot.SnapshotCodecV3
-        </value>
+      co.cask.cdap.data2.transaction.snapshot.SnapshotCodecV1,
+      co.cask.cdap.data2.transaction.snapshot.SnapshotCodecV2,
+      co.cask.tephra.snapshot.SnapshotCodecV3
```
Comment (poornachandra) [inline cdap-common/src/main/resources/cdap-default.xml]: Missing comma.

**Comment by poornachandra** [comment]
One minor comment, please merge after testing on cluster.

**Comment by rohitsinha54** [comment]
Blocked by this issue: https://issues.cask.co/browse/CDAP-3610
We need to make sure that the right codec is getting picked up by logging it. Will re-open after this issue is resolved.

**Comment by rohitsinha54** [comment]
Verified that SnapshotCodecV4 gets picked up by trying to open the snapshot file through tx-debugger
and it complained

Exception in thread "main" java.lang.IllegalArgumentException: Version 4 of snapshot encoding is not supported
	at co.cask.tephra.snapshot.SnapshotCodecProvider.getCodecForVersion(SnapshotCodecProvider.java:95)
	at co.cask.tephra.snapshot.SnapshotCodecProvider.decode(SnapshotCodecProvider.java:128)
	at co.cask.cdap.data2.transaction.TransactionManagerDebuggerMain.retrieveSnapshot(TransactionManagerDebuggerMain.java:392)
	at co.cask.cdap.data2.transaction.TransactionManagerDebuggerMain.executeViewMode(TransactionManagerDebuggerMain.java:266)
	at co.cask.cdap.data2.transaction.TransactionManagerDebuggerMain.parseArgsAndExecMode(TransactionManagerDebuggerMain.java:184)
	at co.cask.cdap.data2.transaction.TransactionManagerDebuggerMain.execute(TransactionManagerDebuggerMain.java:658)
	at co.cask.cdap.data2.transaction.TransactionManagerDebuggerMain.main(TransactionManagerDebuggerMain.java:702)



Merging this PR.

**Pull request merged**